bin/export-codePost-grades

```diff
@@ -0,0 +1,254 @@
+parser.add_argument('-n',
+                    help='Usernames of students to export (by default, everybody).', nargs='*')
+parser.add_argument('--include-inactive', action='store_true',
+                    help='Also export grades of inactive students.')
+parser.add_argument('--json', action='store_true',
+                    help='Export as JSON.')
+parser.add_argument('--pretty', action='store_true',
+                    help='Pretty print output.')
+parser.add_argument('--blackboard', action='store_true',
+                    help='Insert Blackboard column IDs when available (in configuration file, the "lms_ids" option).')
+parser.add_argument('--show-empty', action='store_true',
+                    help='Include columns even of assignments without grades.')
+parser.add_argument('--verbose', action='store_true',
+                    help='Display informational messages.')
+
+
+def process_command_line(parser):
+    global verbose
+
+    args, unknown = parser.parse_known_args()
+    params = vars(args)
+
+    if params["verbose"] == None or params["verbose"] == False:
+        verbose = False
+
+    outparams = params.__repr__()
+    if len(outparams) > 500:
+        outparams = outparams[:500] + "..."
+    _print_info("Command line parameters: {}".format(outparams))
+
+    # Load YAML configuration file
+
+    # Resolve possible locations for this configuration file
+    # FIXME: This is hard-coded, whereas it should be a (documented) constant.
+    possible_locations = ["codepost-config.yaml", ".codepost-config.yaml",
+                          "~/codepost-config.yaml", "~/.codepost-config.yaml"]
+
+    location = None
+    for p in possible_locations:
+        path = os.path.abspath(os.path.expanduser(p))
+        if os.path.exists(path):
+            location = path
+            break
+
+    if location == None:
+        _print_err(
+            "No codepost-config.yaml configuration file detected. Exiting.",
+            fatal=8)
+    else:
+        _print_info("Configuration path: {}".format(location))
+
+    config = load(open(location), Loader=Loader)
+    _print_info("Configuration content: {}".format(config))
+
+    return params, config
+
+
+clparams, config = process_command_line(parser)
+
+grades = cP.get_course_grades(
+    api_key=config["api_key"],
+    course_name=config["course_name"],
+    course_period=config["course_period"])
+
+# JSON output
+if clparams["json"]:
+    import json
+    if clparams["pretty"]:
+        print(json.dumps(grades, indent=2))
+    else:
+        print(json.dumps(grades))
+    sys.exit(0)
+
+
+course = cP.get_course_roster_by_name(
+    api_key=config["api_key"],
+    course_name=config["course_name"],
+    course_period=config["course_period"])
+
+# CSV output
+
+# Figure out headers
+headers = clparams.get("a", list())
+if headers == None or len(headers) == 0:
+    headers = []
+    for aid in course["assignments"]:
+        assignment_info = cP.get_assignment_info_by_id(
```

Comment:
Interesting that you should mention this. I had (mistakenly) remembered that we already had a sorting key, "orderKey" but then realized it was only for the submission queue.

I think that a top feature request is allowing admins to define some ordering of assignments. Typically this is done by due dates, but since we (currently) don't require due dates, we don't have such a stable metadata to sort by.

- I think requiring a due date; or even just optionally accepting a due date as a prerequisite to sorting, might be annoying. I have not yet seen a good UI to enter dates in a form. I think entering dates is something unpleasant, and I think that since people at Princeton (and possibly elsewhere) already enter them for the submission platform, Tigerfile, they would resent having to enter them again.

- I think the best way to address this would be to allow admins to drag-drop-reorder the assignments in one (or all?) of the views, and then use this to determine the "sortKey" type property.

Thoughts?